src/source/tile.js

```diff
@@ -413,17 +413,19 @@ class Tile {
     }
 
     setFeatureState(states: LayerFeatureStates) {
-        if (!this.latestRawTileData || Object.keys(states).length === 0) return;
-
-        if (!this.vtLayers) {
-            this.vtLayers = new vt.VectorTile(new Protobuf(this.latestRawTileData)).layers;
+        if (!this.latestFeatureIndex ||
+            !this.latestRawTileData ||
```

Comment:
Does it need to check for `this.latestRawTileData` since `this.latestFeatureIndex` has its own copy of the raw tile data